src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { FlexContainer } from '../FlexContainer';
+import { FilterRow } from './FilterRow';
+import { DataPointPropTypes, FiltersProps } from './Filters.types';
+import { Operators } from './Filters.enums';
+
+const generateId = ({ operator, dataPoint, condition, input }, index) =>
+  `${operator}-${dataPoint}-${condition}-${input}-${index}`;
```

Comment:
Not perfect but it seems to me be better than let user define id and add another property. Props values should be `string` or `undefined`.